### ISSUE TITLE
Failed Lambda function: "Create Zuora Subscription" - shows as succeeded in State machine dashboard

### ISSUE THREAD

**Issue by michaelwmcnamara** [guardian/support-workers#74]
When the Create Zuora Subscription fails, the failure handler is invoked correctly and the State Machine Lambda failed metric is sent.
However in the status diagram for that execution, the status for that step shows as "succeeded", with the only suggestion there was an error hidden behind the "output" tab.
This is very confusing!

![image](https://user-images.githubusercontent.com/7239344/32794773-b87de534-c961-11e7-80ce-1a573a4af92b.png)

![image](https://user-images.githubusercontent.com/7239344/32794791-c3fc1e12-c961-11e7-93af-b16ed2914fd4.png)

![image](https://user-images.githubusercontent.com/7239344/32794842-e5f1f136-c961-11e7-8fa6-e31708365a3e.png)

**Comment by michaelwmcnamara** [comment]
Example is here: https://eu-west-1.console.aws.amazon.com/states/home?region=eu-west-1#/executions/details/arn:aws:states:eu-west-1:865473395570:execution:MonthlyContributionsPROD-B6RNPDZYIJLZ:0019c53e-d043-4c5e-a344-78b3704cb570

**Comment by jacobwinch** [comment]
I agree that this is a bit confusing. But there's an important distinction here between what we consider to be a failure/success and what AWS considers to be a failure/success. 

This status diagram is showing the status of various processes in AWS (in our case lambda executions). So a failure in this context means that something went fundamentally wrong when calling the lambda and AWS could not execute our code correctly e.g. the handler function (entry point for a lambda) could not be found. 

A success (in this context) just means that AWS were able to run the lambda code successfully - _even if_ that code is handling an error case.

For our purposes we should identify a failed sign-up journey when:

a) The FailureHandler was invoked (this means that an error occurred which meant that retrying would not, or did not, succeed e.g. the user's payment method was invalid)
b) The state machine execution failed completely (e.g. we misconfigured a lambda and it wouldn't actually run at all).

Note that even b is not completely clean cut, because if CreateZuoraSubscription actually does take a payment, we would want to tell the user that their sign-up was a success, even though from a technical perspective we could have broken the state machine completely (e.g. tracking lambda is fundamentally broken). 

In this case we'd want the dev team to know something had gone wrong, but not the user.

Let me know if this doesn't make sense and we can discuss it in more detail!
